### PR TITLE
layers: Add VUID 03673

### LINKED ISSUE #2831: VK_BUFFER_USAGE_ACCELERATION_STRUCTURE_BUILD_INPUT_READ_ONLY_BIT_KHR is not validated
Spec says:

VUID-vkCmdBuildAccelerationStructuresKHR-geometry-03673
The buffers from which the buffer device addresses for all of the geometry.triangles.vertexData, geometry.triangles.indexData, geometry.triangles.transformData, geometry.aabbs.data, and geometry.instances.data members of all pInfos[i].pGeometries and pInfos[i].ppGeometries are queried must have been created with the VK_BUFFER_USAGE_ACCELERATION_STRUCTURE_BUILD_INPUT_READ_ONLY_BIT_KHR usage flag

I can create buffers without that flag and the validation layer will not complain.

I expect it to complain if I don't set that flag on my buffers.

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
Closes #2831.

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -6492,6 +6492,66 @@ bool CoreChecks::PreCallValidateCmdBuildAccelerationStructuresKHR(
...
+    // Parameter validation has already checked VUID-03788: !(pGeometries && ppGeometries)
+    std::function<const VkAccelerationStructureGeometryKHR &(uint32_t)> geom_accessor;
+    if (p_geometries) {
+        geom_accessor = [p_geometries](uint32_t i) -> const VkAccelerationStructureGeometryKHR & { return p_geometries[i]; };
+    } else if (pp_geometries) {
+        geom_accessor = [pp_geometries](uint32_t i) -> const VkAccelerationStructureGeometryKHR & {
+            // pp_geometries[i] is assumed to be a valid pointer
+            return *pp_geometries[i];
+        };
+    }
```
Comment (ncesario-lunarg) [inline layers/core_validation.cpp:6533]: I'm sure there's a better/more elegant/more efficient way to get the same effect here...

Comment (Tony-LunarG) [reply layers/core_validation.cpp:6533]: I did spend more time than I'm willing to admit figuring it out.  But I'm not very C++y

File: tests/vklayertests_others.cpp [hunk 2]
```diff
@@ -10874,6 +10874,19 @@ TEST_F(VkLayerTest, ValidateCmdBuildAccelerationStructuresKHR) {
         vkCmdBuildAccelerationStructuresKHR(m_commandBuffer->handle(), 1, &invalid_build_info_khr, &pBuildRangeInfos);
         m_errorMonitor->VerifyFound();
     }
+
```
Comment (Tony-LunarG) [inline tests/vklayertests_others.cpp:10877]: There are a ton of problems with this test, but you didn't add any

**Review by Tony-LunarG — APPROVED** [review]
LGTM

File: tests/vklayertests_others.cpp (hunk 2 above)
Comment (ncesario-lunarg) [reply tests/vklayertests_others.cpp:10877]: 😬 well, I have an RT card now, so I guess I have no more excuses to not fix them ;)

**Pull request merged**